eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +146,48 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | Unlike @'NominalDiffTime'@ that supports @'Pico'@ precision, this type
+-- only supports @'Micro'@ precision.
+newtype NominalDiffTimeMicro = NominalDiffTimeMicro Micro
+  deriving (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+  deriving newtype (Ord, Num, Fractional, Real, ToJSON, FromJSON, ToCBOR, FromCBOR)
+
+-- | There is no loss of resolution in this conversion
+microToPico :: Micro -> Pico
+microToPico micro = fromRational @Pico $ toRational micro
+
+-- | Loss of resolution occurs in this conversion
+picoToMicro :: Pico -> Micro
+picoToMicro pico = fromRational @Micro $ toRational pico
+
+fromNominalDiffTimeMicro :: NominalDiffTimeMicro -> NominalDiffTime
+fromNominalDiffTimeMicro =
+  secondsToNominalDiffTime . microToPico . nominalDiffTimeMicroToMicroseconds
+
+toNominalDiffTimeMicroWithRounding :: NominalDiffTime -> NominalDiffTimeMicro
+toNominalDiffTimeMicroWithRounding =
+  microsecondsToNominalDiffTimeMicro . picoToMicro . nominalDiffTimeToSeconds
+
+toNominalDiffTimeMicro :: NominalDiffTime -> Maybe NominalDiffTimeMicro
+toNominalDiffTimeMicro ndt
+  | fromNominalDiffTimeMicro ndtm == ndt = Just ndtm
+  | otherwise = Nothing
+  where
+    ndtm = toNominalDiffTimeMicroWithRounding ndt
+
+microsecondsToNominalDiffTimeMicro :: Micro -> NominalDiffTimeMicro
+microsecondsToNominalDiffTimeMicro = NominalDiffTimeMicro
+
+secondsToNominalDiffTimeMicro :: Pico -> NominalDiffTimeMicro
+secondsToNominalDiffTimeMicro = NominalDiffTimeMicro . picoToMicro
```

Comment:
Having both of these functions is confusing and unnecessary. Because:
```haskell
> secondsToNominalDiffTimeMicro 15 == microsecondsToNominalDiffTimeMicro 15
True
```

I suggest keep only one of them and rename it:
```haskell
microToNominalDiffTimeMicro :: Micro -> NominalDiffTimeMicro
microToNominalDiffTimeMicro = NominalDiffTimeMicro
```
Alternative name for the above function could also be `secondsToNominalDiffTimeMicro`, which would stay inline with the previous version. Point is that when you call these functions above you don't supply microseconds or picoseconds, but rather seconds with microsecond or picosecond precision respectively. So, feel free to choose whichever name you like better, the important part is not to keep two functions with morally conflicting names.

Reply:
> when you call these functions above you don't supply microseconds or picoseconds, but rather seconds

Thanks for that! :100: :bow: I completely failed at understanding this :heart: